astropy/io/fits/hdu/image.py

```diff
@@ -245,7 +282,26 @@ def data(self, data):
         else:
             raise ValueError('not a valid data array')
 
+        # Update the header, including adding BZERO/BSCALE if new data is
+        # unsigned. Does not change the values of self._bitpix,
+        # self._orig_bitpix, etc.
         self.update_header()
+        if (data is not None and was_unsigned):
+            self._update_header_scale_info(data.dtype)
```

Comment:
If anything, for unsigned data BZERO and BSCALE should be *kept* in the header, because that's the only way the unsigned ints can be meaningfully represented in FITS.  So in that case keeping them in the header accurately reflects the data (the only reason they are deleted in the generic case is that once the scale factors are applied to the data, they are no longer needed in the header to represent the data unless the data is scaled back before writing).